bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteFlag.java

```diff
@@ -57,10 +77,20 @@
      * @return a set of flags
      */
     public static EnumSet<WriteFlag> getWriteFlags(int flagValue) {
+        EnumSet<WriteFlag> writeFlags = EnumSet.noneOf(WriteFlag.class);
         if ((flagValue & DEFERRED_SYNC.value) == DEFERRED_SYNC.value) {
-            return ONLY_DEFERRED_SYNC;
+            writeFlags.add(DEFERRED_SYNC);
+        }
+        if ((flagValue & DIGEST_TYPE_DUMMY.value) == DIGEST_TYPE_DUMMY.value) {
```

Comment:
Btw what if in this 'flagValue' int multiple digesttypes are enabled? How do we handle this error scenario?

Reply:
We cannot have multiple digestTypes being represented by a single int value. The server initializes the digestType based on the first ever flag value received and doesn't allow it to get updated (unless it was disabled earlier and then enabled). 

We would essentially rely on client to ensure that it doesn't get corrupted. The writeFlags would be automatically set based on digestType. If the client re-opens a writeHandle again (for example, replication), then those bits would be automatically set based on metadata.